CRFModuleValidation/MyJourneyViewController.swift

```diff
@@ -0,0 +1,781 @@
+    
+    @IBOutlet var preferredSizeView: UIView!
+    @IBOutlet var firstDayHeaderView: UIView!
+    @IBOutlet var tableView: UITableView!
+    @IBOutlet var footerView: MyJourneyFooterView!
+    
+    lazy var scheduledActivityManager : MasterScheduledActivityManager = {
+        return MasterScheduledActivityManager(delegate: self)
+    }()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // Hide the views on first load
+        self.firstDayHeaderView.isHidden = true
+        self.tableView.isHidden = true
+    }
+    
+    
+    // MARK: Load and reload - view refresh
+    
+    var isVisible = false
+    var shouldScrollToToday = false
+    var shouldAnimateCompletion = false
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        debugPrint("-- MyJourneyViewController.viewWillAppear")
+        
+        // Set self as the delegate for the schedule manager to listen for changes
+        scheduledActivityManager.delegate = self
+        
+        // Reload on view will appear
+        self.reloadFinished(nil)
+        
+        // Fire request to update the schedules (this will be ignored if already running)
+        scheduledActivityManager.reloadData()
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        
+        isVisible = true
+        if shouldScrollToToday {
+            scrollToTodayIfNeeded()
+        }
+        
+        // TODO emm 2017-11-11 decide if we need this
+//        if let deepLinkTaskGroup = scheduledActivityManager.deepLinkTaskGroup,
+//            let deepLinkTaskId = deepLinkTaskGroup.taskIdentifiers.first {
+//            // We should show a loading dialog to convey to the user that we are
+//            // automatically sending them somewhere after everything has loaded
+//            self.showLoadingView()
+//            scheduledActivityManager.notifyWhenTaskIsAvailable(taskId: deepLinkTaskId.rawValue, callback: { [weak self] (taskId) in
+//                self?.hideLoadingView()
+//                self?.sendUserToDeepLinkTasGroup()
+//            })
+//        }
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        isVisible = false
+    }
+    
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        
+        let headerView = self.tableView.tableHeaderView!
+        headerView.frame = self.firstDayHeaderView.bounds
+        self.tableView.tableHeaderView = headerView
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+    
+    func getLatestSchedules() -> [ScheduleSection] {
+        // TODO: syoung 06/27/2017 Add UI to truncate the schedule if the future is dim.
```

Comment:
You can probably remove the TODO comment.  :)

Reply:
ya think? ;)